lib/fiware-iotagent-lib.js

```diff
@@ -116,6 +130,19 @@ function doActivate(newConfig, callback) {
 
     logger.info(context, 'Activating IOT Agent NGSI Library.');
 
+    if (newConfig.authentication && newConfig.authentication.enabled) {
+        if (newConfig.authentication.type &&
+            newConfig.authentication.type === 'oauth2') {
+            logger.info(context, 'Using OAuth2 as authentication service');
+
+            securityService = require('./services/common/securityServiceOAuth2');
```

Comment:
We typically set require() at the beginning of the file. Thus, instead of ursing require() in this point (and L140) use the corresponding var.

Except if there is a strong reason to have the require defined in the middle...

Reply:
i think i did in the same way it was done for dynamic linking of registries. let me check

Reply:
see: https://github.com/telefonicaid/iotagent-node-lib/blob/e542d79f94ad2a5b129bd817b85d9b7c30ddbb39/lib/fiware-iotagent-lib.js#L124